nengo/builder/signal.py

```diff
@@ -264,6 +272,18 @@ def sparse(self):
         """(bool) Whether the signal is sparse."""
         return is_sparse(self.initial_value)
 
+    def check_for_nans(self):
+        if self.allow_nan:
+            return
```

Comment:
This if clause is redundant. The function will only be invoked if `allow_nan` is `False` except if someone calls it externally (in which case it is debatable which behaviour would be expected).

Reply:
It's only invoked in one place now, but it could be used elsewhere later. But the expected behavior is a little ambiguous, so I'm going to change how it works.